backend/src/main/java/com/soen341/instagram/service/impl/RegistrationService.java

```diff
@@ -0,0 +1,62 @@
+package com.soen341.instagram.service.impl;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.regex.Pattern;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import com.soen341.instagram.dao.impl.AccountRepository;
+import com.soen341.instagram.exception.account.AccountExistException;
+import com.soen341.instagram.exception.account.InvalidEmailException;
+import com.soen341.instagram.model.Account;
+
+@Service("registrationService")
+public class RegistrationService
+{
+	@Autowired
+	private AccountRepository accountRepository;
+
+	@Autowired
+	private PasswordEncoder passwordEncoder;
+
+	public void createNewAccount(final Account account) throws ParseException
+	{
+		if (accountRepository.findByUsername(account.getUsername()) == null
```

Comment:
Does this take into account casing of the username? Will a username with different casing count as different?

Reply:
Database should be case insensitive by default, so this should be ok